lib/stable_email.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+
+# This library is used to parse emails from a public-inbox feed. It is used
+# specifically for the linux stable mailing list
+
+import email
+import email.policy
+
+import git
+
+
+def commit_to_email_message(commit):
+    raw_msg = commit.tree["m"].data_stream.read()
+    m = email.message_from_bytes(raw_msg, policy=email.policy.EmailPolicy(utf8=True))
+    return m
+
+
+def is_greg_request(commit):
+    if commit.author.email == "gregkh@linuxfoundation.org":
+        m = commit_to_email_message(commit)
+        if "X-KernelTest-Branch" in m and "in-reply-to" not in m:
+            return m
+    return False
+
+
+def is_beyond_time_search(commit, limit):
+    if commit.committed_datetime < limit:
+        return True
+    return False
+
+
+def get_version(m):
+    sub = m["subject"]
+    if not sub.endswith("-stable review"):
+        return False
+    if not sub.startswith("["):
+        return False
+    sub1 = sub.replace("-stable review", "")
+    return sub1[sub1.rfind(" ") + 1 :]
+
+
+def get_review_requests(dt_limit):
+    print(
+        "* Looking for review requests after %s..."
+        % dt_limit.strftime("%Y-%m-%d %H:%M UTC")
+    )
+    fg = {}
+    x = 0
+    repo = git.Repo("/data/lore-stable")
+    commits = repo.iter_commits("HEAD")
+
+    for commit in commits:
+        # Limit search
+        if is_beyond_time_search(commit, dt_limit):
+            print("Done. Found %d review requests in %d messages." % (len(fg), x))
+            break
+
+        # Look for Greg's stable RC review requests
+        msg = is_greg_request(commit)
+        if msg:
+            print("Found: %s" % commit.summary)
+            fg[msg["message-id"]] = {"request": commit}
+        x += 1
+
+    return fg
+
+
+def get_review_replies(oldest, fg):
```

Comment:
git_review_requests() and get_review_replies() are very similar. Any opportunities to DRY?